packages/block-library/src/button/edit.native.js

```diff
@@ -327,7 +327,7 @@ function ButtonEdit( props ) {
 	function onRemove() {
 		const { onDeleteBlock, onReplace } = props;
 
-		if ( numOfButtons === 1 ) {
+		if ( numOfButtons === 1 && onDeleteBlock ) {
```

Comment:
It looks like we missed passing this prop in a recent refactor, this `onDeleteBlock` prop should be added [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/block.native.js#L336).